DevTools/HackStudio/LanguageServers/Cpp/CMakeLists.txt

```diff
@@ -0,0 +1,14 @@
+
+compile_ipc(CppLanguageServer.ipc CppLanguageServerEndpoint.h)
+compile_ipc(CppLanguageClient.ipc CppLanguageClientEndpoint.h)
+
+set(SOURCES
+    ClientConnection.cpp
+    main.cpp
+    CppLanguageServerEndpoint.h
+    CppLanguageClientEndpoint.h
+    AutoComplete.cpp
+)
+
+serenity_bin(CppLanguageServer)
+target_link_libraries(CppLanguageServer LibIPC LibGUI)
```

Comment:
How come a language server needs LibGUI? 😨

Reply:
Yeah it is a weird dependency - It's because we need `GUI::CppLexer`.
If we want to not depend on LibGUI here we could either:
1. Create `LibCpp`, similarly to the way we have the javascript lexer in `LibJS`. 
2. Move the c++ lexer to the language server itself, and make `GUI::SyntaxHighlighter` able to go through the c++ language server for lexing.
It would mean that we'd have to use the c++ language server from the TextEditor as well however, which seems wrong.

Reply:
If you decide to continue linking to LibGUI, I would at least expect a comment here explaining the same thing you just said :slightly_smiling_face:

Reply:
Actually there's another dependency we need from LibGUI, the `GUI::TextDocument`.
The language server uses it to update the content of a file when it receives updates over IPC on edit actions the user performs.
We could extract document-related logic to something like `LibDocument`, but it's a pretty big refactoring that I'll leave for a later time.
I ended up extracting the lexer to `LibCpp`, though.